marlowe-integration/README.md

```diff
@@ -13,18 +13,42 @@ following programs need to be in the `PATH` when you use them:
 - `sqitch`
 
 Additionally, you are required to have a PostgreSQL server running and
-available. You can configure the connection settings with the following
+available.
+
+### Setting up a PostgreSQL database (NixOS)
+
+To set up a PostgreSQL database in NixOS, you can use the `services.postgresql` option in your NixOS configuration. Here's an example:
+
+```nix
+services.postgresql = {
+  enable = true;
+  package = pkgs.postgresql;
+  enableTCPIP = true;
+  authentication = pkgs.lib.mkOverride 10 ''
+    local all all trust
+    host all all ::1/128 trust
+    host all all 127.0.0.1/32 trust
+  '';
+};
+```
+
+This configuration enables the PostgreSQL service, installs the `postgresql` package, and allows connections from localhost (`127.0.0.1`) using the default user `postgres` which starts with no password. The `authentication` option adds entries to the `pg_hba.conf` file to allow the integration tests to connect locally.
+
+### Customizing the PostgreSQL configuration
```

Comment:
These instructions fail for me, even though `psql` can connect to the database with those same environment variables.

```console
  test/Language/Marlowe/Runtime/Integration/Basic.hs:32:8: 
  1) Language.Marlowe.Runtime.Integration, Marlowe runtime API, Basic e2e scenario
       uncaught exception: TestnetException
       TestnetException:
         exception: 'user error (Fatal database error: Just "no connection to the server\n")'
         workspace: /tmp/workspaces/local-testnet-cec868d8-61b5-48b5-a5df-5bf354e836f4
Randomized with seed 24446332

Finished in 18.9951 seconds
633 examples, 3 failures, 630 pending

$ PGPASS=$MARLOWE_RT_TEST_DB_PASSWORD psql -h $MARLOWE_RT_TEST_DB_HOST -p $MARLOWE_RT_TEST_DB_PORT -U $MARLOWE_RT_TEST_DB_USER
psql (14.5, server 13.9)
Type "help" for help.

cardano=> \quit
```

Reply:
Huh. Are you using `127.0.0.1` as host?
If so, what does the PostgreSQL log say?
`journalctl -u postgresql | tail`

Reply:
```console
$ journalctl -u postgresql
-- No entries --
```

Reply:
So it works if I do a `createdb -O $MARLOWE_RT_TEST_DB_USER $MARLOWE_RT_TEST_TEMP_DB`, but it doesn't work if `$MARLOWE_RT_TEST_TEMP_DB` hasn't been created.

This seems odd because I thought the tests create the database and that the user doesn't have to create it manuallly.

Reply:
In summary, to get this to work I needed to do the following:
- Use `export` for the environment variables.
- Make sure that `$MARLOWE_RT_TEST_TEMP_DB` is created and owned by `$MARLOWE_RT_TEST_DB_USER`.

Reply:
Ah, that make sense, still strange it says "no connection to the server". The `MARLOWE_RT_TEST_TEMP_DB` does have to exist because it is a template. Apparently `template0` and `template1` are provided by postgresql by default